modules/appeals_api/lib/appeals_api/token_validation_client.rb

```diff
@@ -42,7 +53,16 @@ def validate_token!(audience:, token:, scopes:)
 
       matching_scope = scopes.find { |scope| permitted.include?(scope) }
 
-      raise ::Common::Exceptions::Forbidden unless matching_scope
+      raise ::Common::Exceptions::Forbidden if matching_scope.blank?
+
+      body = JSON.parse(response.body)
+
+      is_veteran_token = permitted.any? { |s| s.start_with? 'veteran/' }
```

Comment:
It would not make sense to receive a token with scopes for both a veteran and a representative, or both system and veteran scopes, so I think this logic is ok